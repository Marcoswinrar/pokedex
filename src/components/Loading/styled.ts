import styled, { keyframes } from 'styled-components'

export const spinner = keyframes`
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
`

export const Container = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  min-height: 70vh;
`

export const Spinner = styled.div`
  width: 50px;
  height: 50px;
  border: 10px solid #f3f3f3; /* Light grey */
  border-top: 10px solid #383636; /* Black */
  border-radius: 50%;
  animation: ${spinner} 1.5s linear infinite;
`
