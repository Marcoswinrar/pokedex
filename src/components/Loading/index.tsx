import * as S from './styled'

const Loading = () => (
  <S.Container>
    <S.Spinner />
  </S.Container>
)

export default Loading
