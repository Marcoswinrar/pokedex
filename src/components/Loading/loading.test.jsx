import { render, screen } from '@testing-library/react'
import Loading from '.'

describe('Componente <Loading/>', () => {
  it('Deve renderizar o componente sem problemas', () => {
    const { container } = render(<Loading />)

    expect(container).toMatchSnapshot()
  })
})
