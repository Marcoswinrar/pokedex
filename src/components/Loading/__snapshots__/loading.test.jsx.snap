// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente <Loading/> Deve renderizar o componente sem problemas 1`] = `
<div>
  <div
    class="sc-bczRLJ eAA-DkR"
  >
    <div
      class="sc-gsnTZi jXfiJV"
    />
  </div>
</div>
`;
